tests/compiler/test_conditional.py

```diff
@@ -245,3 +245,63 @@ def test_if_relation_count_false(snippetcompiler):
     (_, scopes) = compiler.do_compile()
     root: Namespace = scopes.get_child("__config__")
     assert not root.lookup("t").get_value().lookup("multiple").get_value()
+
+
+def test_1573_if_dict_lookup(snippetcompiler):
+    snippetcompiler.setup_for_snippet(
+        """
+entity Test:
+    bool t_success
+    bool f_success
+end
+
+implement Test using std::none
+
+dct = {"t": true, "f": false}
+
+x = Test()
+if dct["t"]:
+    x.t_success = true
+else:
+    x.t_success = false
+end
+
+if dct["f"]:
+    x.f_success = false
+else:
+    x.f_success = true
+end
+        """,
+    )
+    (_, scopes) = compiler.do_compile()
+    root: Namespace = scopes.get_child("__config__")
+    assert root.lookup("x").get_value().lookup("t_success").get_value() is True
+    assert root.lookup("x").get_value().lookup("f_success").get_value() is True
+
+
+def test_1573_implementation_condition_dict_lookup(snippetcompiler):
```

Comment:
add a testcase with a non-boolean value in the dict

Reply:
Could you expand on this? Because of #1808 this might not produce the expected result. Doesn't such a test belong in a PR for that issue?

Reply:
Correct, this undesired behavior was pre-existing, I was not aware of that.